compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala

```diff
@@ -28,54 +28,95 @@ import scala.collection.concurrent.TrieMap
 import scala.jdk.CollectionConverters._
 import scala.util.control.NonFatal
 
-object ScenarioServiceMain extends App {
+private final case class ScenarioServiceConfig(
+    maxInboundMessageSize: Int,
+    enableScenarios: Boolean,
+)
+
+private object ScenarioServiceConfig {
   // default to 128MB
-  val maxMessageSize = args.headOption.map(_.toInt).getOrElse(128 * 1024 * 1024)
-
-  // Needed for the akka Ledger bindings used by Daml Script.
-  val system = ActorSystem("ScriptService")
-  implicit val sequencer: ExecutionSequencerFactory =
-    new AkkaExecutionSequencerPool("ScriptServicePool")(system)
-  implicit val materializer: Materializer = Materializer(system)
-  implicit val ec: ExecutionContext = system.dispatcher
-
-  val server =
-    NettyServerBuilder
-      .forAddress(new InetSocketAddress(InetAddress.getLoopbackAddress, 0)) // any free port
-      .addService(new ScenarioService())
-      .maxInboundMessageSize(maxMessageSize)
-      .build
-  server.start()
-  // Print the allocated port for the client
-  println("PORT=" + server.getPort.toString)
-
-  // Bump up the log level
-  Logger.getLogger("io.grpc").setLevel(Level.ALL)
-
-  // Start a thread to watch stdin and terminate
-  // if it closes. This makes sure we do not leave
-  // this process running if the parent exits.
-  new Thread(new Runnable {
-    def run(): Unit = {
-      while (System.in.read >= 0) {}
-      System.err.println("ScenarioService: stdin closed, terminating server.")
-      server.shutdown()
-      system.terminate()
-      ()
-    }
-  }).start()
+  val DefaultMaxInboundMessageSize: Int = 128 * 1024 * 1024
+
+  private class OptionParser()
```

Comment:
please let me know if this looks okay!

Reply:
```suggestion
  private class OptionParser
```

Reply:
instead of making this a class you can also make it a `val` in the object. I tend to go for the latter usually but both are perfectly fine.

Reply:
hm, I was trying to follow `triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceConfig.scala` as closely as possible, that one defines its `OptionParser` as a class, though that one takes an argument. Is that the difference?

Reply:
yeah with the value you cannot take an argument and you cannot extend it so it’s a bit more restrictive and arguably clearer in intent if those restrictions apply. But really just a very minor stylistic choice. What you have is perfectly fine.

Reply:
thanks!